src/tpl.c

```diff
@@ -302,7 +306,7 @@ char *calc_field_addr(tpl_node *parent, int type,char *struct_addr, int ordinal)
         align_sz = tpl_types[type].sz;
         break;
     }
-    offset = ((uintptr_t)prev->addr - (uintptr_t)struct_addr)
+    offset = ((size_t)prev->addr - (size_t)struct_addr)
```

Comment:
I rather see the ptrdiff_t type here. See https://www.viva64.com/en/a/0050/ for example.